modules/src/core/ics03_connection/handler/conn_open_confirm.rs

```diff
@@ -1,18 +1,19 @@
 //! Protocol logic specific to processing ICS3 messages of type `MsgConnectionOpenConfirm`.
 
+use crate::clients::host_functions::HostFunctionsProvider;
 use crate::core::ics03_connection::connection::{ConnectionEnd, Counterparty, State};
-use crate::core::ics03_connection::context::ConnectionReader;
 use crate::core::ics03_connection::error::Error;
 use crate::core::ics03_connection::events::Attributes;
 use crate::core::ics03_connection::handler::verify::verify_proofs;
 use crate::core::ics03_connection::handler::{ConnectionIdState, ConnectionResult};
 use crate::core::ics03_connection::msgs::conn_open_confirm::MsgConnectionOpenConfirm;
+use crate::core::ics26_routing::context::ReaderContext;
 use crate::events::IbcEvent;
 use crate::handler::{HandlerOutput, HandlerResult};
 use crate::prelude::*;
 
-pub(crate) fn process(
-    ctx: &dyn ConnectionReader,
+pub(crate) fn process<HostFunctions: HostFunctionsProvider>(
+    ctx: &dyn ReaderContext,
```

Comment:
Why is it important to use the `ReaderContext` here? It seems like `ConnectionReader` was good; it's unclear which changes in the `process` handler warranted switching to `ReaderContext`.

Reply:
We didn't like the duplicate methods in different context traits, for example `cient_consensus_state` appearing in `ClientReader`, `ChannelReader` and `ConnectionReader`, so we removed them, and created the `ReaderContext` super trait to capture all that is needed to read anything from the store.

Reply:
Plus our beefy light client needed some methods that were not available on the `ConnectionReader`